PangleAdapter/src/main/java/com/chartboost/mediation/pangleadapter/PangleAdapter.kt

```diff
@@ -347,103 +403,72 @@ class PangleAdapter : PartnerAdapter {
         request: PartnerAdLoadRequest,
         partnerAdListener: PartnerAdListener
     ): Result<PartnerAd> {
         return suspendCoroutine { continuation ->
-            val bannerAd = TTAdSdk.getAdManager().createAdNative(context)
-
-            val adSlot = request.size?.let {
-                val widthDp = it.width.toFloat()
-                val heightDp = it.height.toFloat()
-
-                PartnerLogController.log(
-                    CUSTOM,
-                    "Pangle setting banner with size (w: $widthDp, h: $heightDp)"
-                )
-
-                AdSlot.Builder()
-                    .setCodeId(request.partnerPlacement)
-                    .setExpressViewAcceptedSize(
-                        widthDp,
-                        heightDp
-                    ).build()
-            }
-
-            bannerAd.loadBannerExpressAd(adSlot, object : TTAdNative.NativeExpressAdListener {
-                override fun onError(code: Int, message: String?) {
-                    PartnerLogController.log(LOAD_FAILED, "Code: $code. Error: $message")
-                    continuation.resume(
-                        Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN))
-                    )
-                }
+            PAGBannerAd.loadAd(
+                request.partnerPlacement,
+                PAGBannerRequest(getPangleBannerSize(request.size)),
+                object : PAGBannerAdLoadListener {
+                    override fun onError(code: Int, message: String?) {
+                        PartnerLogController.log(LOAD_FAILED, "Code: $code. Error: $message")
+                        continuation.resume(
+                            Result.failure(ChartboostMediationAdException(ChartboostMediationError.CM_LOAD_FAILURE_UNKNOWN))
+                        )
+                    }
 
-                override fun onNativeExpressAdLoad(bannerAds: MutableList<TTNativeExpressAd>?) {
-                    bannerAds?.firstOrNull()?.let { banner ->
+                    override fun onAdLoaded(pagBannerAd: PAGBannerAd?) {
                         PartnerLogController.log(LOAD_SUCCEEDED)
-
-                        banner.setExpressInteractionListener(object :
-                            TTNativeExpressAd.ExpressAdInteractionListener {
-                            override fun onAdClicked(bannerView: View?, type: Int) {
-                                PartnerLogController.log(DID_CLICK)
-                                partnerAdListener.onPartnerAdClicked(
-                                    PartnerAd(
-                                        ad = bannerView,
-                                        details = emptyMap(),
-                                        request = request
+                        pagBannerAd?.let {
+                            it.setAdInteractionListener(object : PAGBannerAdInteractionListener {
+                                override fun onAdShowed() {
```

Comment:
didn't we say to not infer impression events in the case that the underlying network doesn't have an actual impression callback?

Reply:
They used to have a callback called `onAdShow`. This new one is now in past tense, but yes we did say to not infer impression events unless there's an actual impression callback. Let me double-check, in case. But most likely, we won't include this.

Reply:
@bichenwang this is what their docs mention. Will remove it from our end, if Pangle can later confirm for us to add the impression here, we can do it then.

<img width="793" alt="image" src="https://github.com/ChartBoost/chartboost-mediation-android-adapter-pangle/assets/30056771/9e99d8a4-c6d9-4881-88f1-966854e5e33f">